api/infoGatherer/specDetails.go

```diff
@@ -169,52 +184,61 @@ func (s *SpecInfoGatherer) getParsedConcepts() map[string]*gauge.Concept {
 }
 
 func (s *SpecInfoGatherer) getStepsFromCachedSpecs() map[string][]*gauge.StepValue {
+	s.specsCache.mutex.RLock()
```

Comment:
Should probably stick to one paradigm. Elsewhere there is an unlock call using `defer`.